File: C/binary search/meta-llama-llama-4-maverick-17b-128e-instruct.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

// Comparison function for qsort
int compare(const void* a, const void* b) {
    return (*(int*)a - *(int*)b);
}

// Function to perform Binary Search
int binarySearch(int arr[], int target, int low, int high) {
    while (low <= high) {
        int mid = low + (high - low) / 2;

        // Check if target is present at the mid
        if (arr[mid] == target)
            return mid;

        // If target is greater, ignore left half
        if (arr[mid] < target)
            low = mid + 1;

        // If target is smaller, ignore right half
        else
            high = mid - 1;
    }

    // If we reach here, then the element was not present
    return -1;
}

// Function to generate a sorted array of 'n' random integers
int* generateRandomSortedArray(int n) {
    int* arr = (int*)malloc(n * sizeof(int));
    for (int i = 0; i < n; i++) {
        arr[i] = rand() % 10000; // Generate random numbers between 0 and 9999
    }
    // Sort the array in ascending order
    qsort(arr, n, sizeof(int), compare);
    return arr;
}

int main() {
    FILE* f = fopen("../../inputs.txt", "r");
    if (!f) {
        return 1;
    }

    int capacity = 1000;
    int* array = (int*)malloc(capacity * sizeof(int));
    int size = 0;

    while (fscanf(f, "%d", &array[size]) == 1) {
        size++;
        if (size >= capacity) {
            capacity *= 2;
            int* temp = realloc(array, capacity * sizeof(int));
            if (!temp) {
                free(array);
                fclose(f);
                return 1;
            }
            array = temp;
        }
    }
    fclose(f);

    // Ordena o array antes da busca binária
    qsort(array, size, sizeof(int), compare);

    int target = 42;  // Valor a ser buscado
    int index = binarySearch(array, target,0, size-1);

    free(array);
    return 0;
}

File: Java/binary search/MetaLlamaLlama4Maverick17b128eInstruct.java
public class MetaLlamaLlama4Maverick17b128eInstruct {

    /**
     * Searches for an element in a sorted array using Binary Search.
     * 
     * @param arr    the sorted array to search in
     * @param target the element to search for
     * @return the index of the target element if found, -1 otherwise
     */
    public static int binarySearch(int[] arr, int target) {
        int left = 0;
        int right = arr.length - 1;
        while (left <= right) {
            int mid = left + (right - left) / 2;

            // Check if target is present at the mid
            if (arr[mid] == target) {
                return mid;
            }

            // If target is greater, ignore left half
            if (arr[mid] < target) {
                left = mid + 1;
            }

            // If target is smaller, ignore right half
            else {
                right = mid - 1;
            }
        }

        // if we reach here, then element was not present
        return -1;
    }

    public static void main(String[] args) {
        // Generate a large sorted array
        int[] arr = new int[1000];
        for (int i = 0; i < 1000; i++) {
            arr[i] = i;
        }

        // Perform binary search
        int target = 512; // Example target
        int result = binarySearch(arr, target);

        // You can use the result as needed, e.g., store it or use it in further computations
    }
}